open_targets/data/_ftp_client.py

```diff
@@ -0,0 +1,89 @@
+"""FTP client module.
+
+This module provide convenient functions for use to list and download files from
+Open Targets Platform FTP server.
+"""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from ftplib import FTP
+from io import BytesIO
+from typing import Any, Concatenate, ParamSpec, TypeVar
+
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+def _ftp_client_connect(
+    func: Callable[Concatenate["FTPClient", FTP, P], R],
+) -> Callable[Concatenate["FTPClient", P], R]:
+    def wrapper(client: "FTPClient", *args: P.args, **kwargs: P.kwargs) -> R:
+        ftp = FTP(client.host)  # noqa: S321
+        ftp.login()
+        result = func(client, ftp, *args, **kwargs)
+        ftp.close()
+        return result
+
+    return wrapper
+
+
+@dataclass(frozen=True)
+class FTPClientListResult:
+    files: list[str]
+    dirs: list[str]
+
+
+class FTPClient:
+    host: str
+
+    def __init__(self, host: str) -> None:
+        self.host = host
+
+    @_ftp_client_connect
```

Comment:
I am getting a pylance warning here.
 ```
 Argument of type "(self: Self@FTPClient, ftp: FTP, path: str) -> FTPClientListResult" cannot be assigned to parameter "func" of type "(FTPClient, FTP, **P@_ftp_client_connect) -> R@_ftp_client_connect" in function "_ftp_client_connect"
  Type "(self: Self@FTPClient, ftp: FTP, path: str) -> FTPClientListResult" is incompatible with type "(FTPClient, FTP, **P@_ftp_client_connect) -> R@_ftp_client_connect"
    Parameter 1: type "FTPClient" is incompatible with type "Self@FTPClient"
      Type "FTPClient" is incompatible with type "Self@FTPClient"
```

Reply:
Really don't know where it comes from, I don't have this issue on my side but I will ignore it for now.